Infinite scroll

far0uq opened the pull request.

far0uq committed:
feat: infinite scrolling along with search by category

Infinite Scrolling implemented for items. Reaching the last item will trigger a GET/POST for more items.

Furthermore, implemented search by Category as well as by query for items.

far0uq committed:
update: re-organizing file structure

far0uq committed:
update: new UI 1.0

Implemented custom themes through ConfigProviders. First attempt to make it responsive.

Made Navbar + ItemContainer + SearchBar responsive.

far0uq committed:
update: introduced reponsive checks and other fixes

Made the ItemsLoading Component responsive. Items changed from having the 'justify' property that made issues in alignment to 'gap' property.

Infinite Scroll fixed where ref was being passed down before.

mohammadahmadteq reviewed (approved):
![emoticon-showing-thumb-up-2BWXA9N-removebg-preview (1)](https://github.com/user-attachments/assets/ee72be63-72db-4806-96f4-c36026c53936)

far0uq committed:
update: consumed design token in itemcontainer and updated readme

far0uq committed:
feat: cart created and cart items slice initialized

Added the cart feature as well as its toggling and incorporated it in the already existing Navbar.

Create the Zustand slice for the cart and configured it to be modular
